### ISSUE TITLE
Invalid foreign key ON UPDATE action specified in schema.rb

### ISSUE THREAD

**Issue by nilbus** [ibmdb/ruby-ibmdb#52]
In our Rails 4.2 app, after foreign key support was added to ibm-db (in our case on ibm-db 3.0.1), running `rake db:migrate` adds foreign key definitions to schema.rb, as expected. However, for our schema, it adds the option `on_update: :cascade`, which is not a legal `ON UPDATE` action.

``` ruby
add_foreign_key "AUDIT_DETAILS", "AUDITS", column: "AUDIT_ID", primary_key: "ID", name: "FK_AUDDTL_ID", on_update: :cascade, on_delete: :restrict
```

What follows is the relevant portion of our DDL that defines this foreign key. Notice `ON DELETE CASCADE` and `ON UPDATE RESTRICT`.

``` sql
ALTER TABLE "MARKETING"."AUDIT_DETAILS"
        ADD CONSTRAINT "FK_AUDDTL_ID" FOREIGN KEY
                ("AUDIT_ID")
        REFERENCES "MARKETING"."AUDITS"
                ("ID")
        ON DELETE CASCADE
        ON UPDATE RESTRICT
        ENFORCED
        ENABLE QUERY OPTIMIZATION;
```

Naturally, when we try to load this new schema.rb, DB2 tells us that `ON UPDATE CASCADE` is indeed illegal.

```
ActiveRecord::StatementInvalid: RuntimeError: Failed to execute statement due to: [IBM][CLI Driver][DB2/LINUXX8664] SQL0104N  An unexpected token "CASCADE" was found following "ON UPDATE".  Expected tokens may include:  "RESTRICT, NO ACTION".  SQLSTATE=42601 SQLCODE=-104: ALTER TABLE AUDIT_DETAILS ADD CONSTRAINT FK_AUDDTL_ID
FOREIGN KEY (AUDIT_ID)
  REFERENCES AUDITS (ID)
 ON DELETE RESTRICT ON UPDATE CASCADE
```

I can work with you to provide any other information you need to reproduce or fix this bug. Thanks!

**Comment by arvindgu** [comment]
Yes, please provide the steps to reproduce this issue so that I can analyze it.

**Comment by nilbus** [comment]
@arvindgu Created an example app that reproduces this behavior, with instructions in the README on how to do it.

https://github.com/nilbus/ibm_db_bug

**Comment by arvindgu** [comment]
@nilbus I was able to reproduce this issue using the example app you provided. We will fix it in next release.

**Comment by arvindgu** [comment]
We have fixed this in ibm_db version 3.0.2. Please try and let me know.

**Comment by arvindgu** [comment]
Closing this, we it is already fixed.
